ValueError: Hash algorithm not known for ed25519

When running the tests with a version of asn1crypto that includes https://github.com/wbond/asn1crypto/pull/265, one of them fails:
```
________________________ test_keyset_templates_in_arch _________________________

    def test_keyset_templates_in_arch():
        cfg = CertomancerConfig.from_file(
            'tests/data/with-arch-templates.yml', 'tests/data'
        )
        old_arch = cfg.get_pki_arch(ArchLabel('testing-ca'))
        algo = old_arch.get_cert(CertLabel('root')).public_key.algorithm
        assert algo == 'rsa'
    
        new_arch = cfg.get_pki_arch(ArchLabel('testing-ca-2'))
        algo = new_arch.get_cert(CertLabel('root')).public_key.algorithm
        assert algo == 'rsa'
    
        newer_arch = cfg.get_pki_arch(ArchLabel('testing-ca-3'))
>       algo = newer_arch.get_cert(CertLabel('root')).public_key.algorithm

tests/test_certs.py:859: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
certomancer/registry/pki_arch.py:952: in get_cert
    signature = generic_sign(
certomancer/crypto_utils.py:315: in generic_sign
    return CRYPTO_BACKEND.generic_sign(private_key, tbs_bytes, signature_algo)
certomancer/crypto_utils.py:183: in generic_sign
    digest_algorithm = sd_algo.hash_algo
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <asn1crypto.algos.SignedDigestAlgorithm 140737282252816 b'0\x05\x06\x03+ep'>

    @property
    def hash_algo(self):
        """
        :return:
            A unicode string of "md2", "md5", "sha1", "sha224", "sha256",
            "sha384", "sha512", "sha512_224", "sha512_256" or "shake256"
        """
    
        algorithm = self['algorithm'].native
        if algorithm in self._algo_map:
            return self._algo_map[algorithm]
    
        if algorithm == 'rsassa_pss':
            return self['parameters']['hash_algorithm']['algorithm'].native
    
        if algorithm == 'ed25519' or algorithm == 'ed448':
>           raise ValueError(unwrap(
                '''
                Hash algorithm not known for %s - use .cms_hash_algorithm for CMS purposes.
                More info at https://github.com/wbond/asn1crypto/pull/230.
                ''',
                algorithm
            ))
E           ValueError: Hash algorithm not known for ed25519 - use .cms_hash_algorithm for CMS purposes. More info at https://github.com/wbond/asn1crypto/pull/230.
```

--- a/certomancer/crypto_utils.py
+++ b/certomancer/crypto_utils.py
@@ -94,11 +94,10 @@ class PycaCryptographyBackend(CryptoBackend):
         priv_key = serialization.load_der_private_key(
             priv_key_bytes, password=None
         )
-        digest_algorithm = sd_algo.hash_algo
         sig_algo = sd_algo.signature_algo
         if sig_algo == 'rsassa_pkcs1v15':
             asym_padding = padding.PKCS1v15()
-            hash_algo = getattr(hashes, digest_algorithm.upper())()
+            hash_algo = getattr(hashes, sd_algo.hash_algo.upper())()
             assert isinstance(priv_key, rsa.RSAPrivateKey)
             return priv_key.sign(tbs_bytes, asym_padding, hash_algo)
         elif sig_algo == 'rsassa_pss':
@@ -118,15 +117,15 @@ class PycaCryptographyBackend(CryptoBackend):
             pss_padding = padding.PSS(
                 mgf=padding.MGF1(algorithm=mgf_md), salt_length=salt_len
             )
-            hash_algo = getattr(hashes, digest_algorithm.upper())()
+            hash_algo = getattr(hashes, sd_algo.hash_algo.upper())()
             assert isinstance(priv_key, rsa.RSAPrivateKey)
             return priv_key.sign(tbs_bytes, pss_padding, hash_algo)
         elif sig_algo == 'dsa':
             assert isinstance(priv_key, dsa.DSAPrivateKey)
-            hash_algo = getattr(hashes, digest_algorithm.upper())()
+            hash_algo = getattr(hashes, sd_algo.hash_algo.upper())()
             return priv_key.sign(tbs_bytes, hash_algo)
         elif sig_algo == 'ecdsa':
-            hash_algo = getattr(hashes, digest_algorithm.upper())()
+            hash_algo = getattr(hashes, sd_algo.hash_algo.upper())()
             assert isinstance(priv_key, ec.EllipticCurvePrivateKey)
             return priv_key.sign(
                 tbs_bytes, signature_algorithm=ec.ECDSA(hash_algo)

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -55,7 +55,7 @@ testing-minimal = [
     "certomancer[requests-mocker,web-api]"
 ]
 testing = [
-    "pyhanko-certvalidator==0.23.0",
+    "pyhanko-certvalidator==0.26.5",
     "certomancer[testing-minimal,pkcs11]"
 ]
 mypy = [

